bluej/src/main/java/bluej/pkgmgr/target/ClassTarget.java

```diff
@@ -1079,46 +1098,38 @@ public SourceType getSourceType()
         return sourceAvailable;
     }
 
-    /**
-     * @return the name of the Java file this target corresponds to. In the case of a Stride target this is
-     *          the file generated during compilation.
-     */
-    public File getJavaSourceFile() {
+    private File getSourceFileByType(SourceType sourceType)
+    {
         if (null == getPackage())
         {
             return null;
         }
-        else
-        {
-            return new File(getPackage().getPath(), getBaseName() + "." + SourceType.Java.toString().toLowerCase());
-        }
+        String sourceFileName = ktSourceFileName.orElse(getBaseName() + "." + sourceType.getExtension());
+        return new File(getPackage().getPath(), sourceFileName);
+    }
+
+    /**
+     * @return the name of the Java file this target corresponds to. In the case of a Stride target this is
+     *          the file generated during compilation.
+     */
+    public File getJavaSourceFile()
+    {
+        return getSourceFileByType(SourceType.Java);
     }
     
     /**
      * @return the name of the Stride file this target corresponds to. This is only valid for Stride targets.
      */
     public File getFrameSourceFile()
     {
-        if (null == getPackage())
-        {
-            return null;
-        }
-        else
-        {
-            return new File(getPackage().getPath(), getBaseName() + "." + SourceType.Stride.toString().toLowerCase());
-        }
+        return getSourceFileByType(SourceType.Stride);
     }
-    
-    @SuppressWarnings("incomplete-switch")
+
     @Override
     public File getSourceFile()
     {
-        switch (sourceAvailable)
-        {
-            case Java: return getJavaSourceFile();
-            case Stride: return getFrameSourceFile();
-        }
-        return null;
+        SourceType sourceType = sourceAvailable == SourceType.Stride ? SourceType.Java : sourceAvailable;
```

Comment:
Would be good to add a comment here explaining that Kotlin is dealt with in getSourceFileByType; that can return a Kotlin file even if SourceType.Java is passed.  (I think?)

Reply:
Actually, this seems to be my mistake. This method should return Stride source, not Java for Stride class targets.